schema/formats.go

```diff
@@ -51,6 +55,41 @@ func (f cidrOrIPv4FormatChecker) IsFormat(input string) bool {
 	return (cidrErr == nil && cidrIP.To4() != nil) || (ip != nil && ip.To4() != nil)
 }
 
+func (f ipv4NetworkFormatChecker) IsFormat(input string) bool {
+	hostIP, netIP, cidrErr := extractHostAndNet(input)
+	if cidrErr != nil {
+		return false
+	}
+	return reflect.DeepEqual(hostIP, netIP)
+}
+
+func (f ipv4AddressWithCidrFormatChecker) IsFormat(input string) bool {
+	hostIP, netIP, cidrErr := extractHostAndNet(input)
+	if cidrErr != nil {
+		return false
+	}
+
+	for i := range hostIP {
+		if hostIP[i] == 255 {
+			return false
+		}
+	}
+	return !reflect.DeepEqual(hostIP, netIP)
+}
+
+func extractHostAndNet(input string) (hostIP net.IP, netIP net.IP, err error) {
```

Comment:
this method loses cidr information, I think if would be called extractHostAndNetAddresses, it might be more to the point.